### ISSUE TITLE
我不同环境报的bug 还不一样，例如这次遇到的是8卡机器，但只使用了1卡

### ISSUE THREAD

**Issue by Syno8** [ssbuild/chatglm_finetuning#56]
很奇怪的是我有8卡，然后跑起来每次用的1个卡

> INFO:lightning_fabric.utilities.seed:[rank: 0] Global seed set to 42
INFO:lightning_fabric.utilities.distributed:Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/1
INFO:torch.distributed.distributed_c10d:Added key: store_based_barrier_key:1 to store for rank: 0
INFO:torch.distributed.distributed_c10d:Rank 0: Completed store-based barrier for key:store_based_barrier_key:1 with 1 nodes.
INFO:pytorch_lightning.utilities.rank_zero:----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 1 processes
----------------------------------------------------------------------------------------------------

> /home/bmp/.local/lib/python3.8/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:613: UserWarning: Checkpoint directory /best_ckpt exists and is not empty.
  rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
INFO:pytorch_lightning.utilities.rank_zero:Loading `train_dataloader` to estimate number of stepping batches.
/path/to/python3.8/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:224: PossibleUserWarning: The dataloader, train_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 128 which is the number of cpus on this machine) in the `DataLoader` initto improve performance.
  rank_zero_warn(
INFO:pytorch_lightning.accelerators.cuda:LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3,4,5,6,7]

**Comment by ssbuild** [comment]
(https://github.com/ssbuild/chatglm_finetuning/blob/88479c885b72044f0d1929a14a817f44d7751ed9/data_utils.py#L25)
修改设备数量

**Comment by Syno8** [comment]
@ssbuild i set 8, it use 1 gpu

**Comment by ssbuild** [comment]
try  cuda_launch_blocking=1 python train.py

**Comment by Syno8** [comment]
我加上以后，并没有显示报错，或提示等新的信息，确实设置的deivce 生效了，通过torch.cuda.device_count() 也是8卡

**Comment by ssbuild** [comment]
根据你的反馈，能给出的建议是升级nccl ，重装deepspeed.

**Comment by paulcx** [comment]
我也曾遇到过这个问题，另外基于deepspeed多卡训练的时候应该如何启动？直接运行python train.py会报错没有获取到rank地址，目前只能用torchrun --nproc_per_node=8 train.py。不太熟悉pytorch-lightning，是不是正确的启动方式？
